src/main/scala/com/fulcrumgenomics/util/UpdateDelimitedFileContigNames.scala

```diff
@@ -0,0 +1,117 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.util
+
+import com.fulcrumgenomics.FgBioDef.{FilePath, PathToSequenceDictionary, _}
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.commons.util.{LazyLogging, StringUtil}
+import com.fulcrumgenomics.fasta.SequenceDictionary
+import com.fulcrumgenomics.sopt.{arg, clp}
+
+@clp(description =
+  """
+    |Updates then contig names in a delimited data file.
+    |
+    |The name of each sequence must match one of the names (including aliases) in the given sequence dictionary.  The
+    |new name will be the primary (non-alias) name in the sequence dictionary.
+  """,
+  group = ClpGroups.Utilities)
+class UpdateDelimitedFileContigNames
+(@arg(flag='i', doc="Input delimited data file.") val input: FilePath,
+ @arg(flag='d', doc="The path to the sequence dictionary with contig aliases.") val dict: PathToSequenceDictionary,
+ @arg(flag='c', doc="The column indices for the contig names (0-based).") val columns: Seq[Int],
+ @arg(flag='o', doc="Output delimited data file.") val output: FilePath,
+ @arg(flag='T', doc="The delimiter") val delimiter: Char = '\t',
+ @arg(flag='H', doc="Treat lines with this starting string as comments (always printed)") val header: String = "#",
+ @arg(doc="Skip missing contigs.") val skipMissing: Boolean = false
+) extends FgBioTool with LazyLogging {
+
+  Io.assertReadable(input)
+  Io.assertReadable(Seq(input, dict))
+  Io.assertCanWriteFile(output)
+
+  this.columns.foreach(column => validate(column >= 0, s"Column index < 0: $column"))
+
+
+  override def execute(): Unit = {
+    val dict   = SequenceDictionary(this.dict)
+    val in     = Io.readLines(input).bufferBetter
+    val out    = Io.toWriter(output)
+
+    val progress = ProgressLogger(logger, noun="lines", verb="written")
+
+    // write the header
+    in.takeWhile(_.startsWith(this.header)).foreach { line =>
+      progress.record()
+      out.write(line); out.write('\n')
+    }
+
+    // write the rest of the lines
+    var fields = Array("", "", "", "")
+    in.foreach { line =>
+      progress.record()
+
+      // Split the line, but support lines that have a variable number of fields
+      val numFields   = StringUtil.split(line, delimiter, fields)
+      val numExpected = {
+        var count: Int = 0
+        forloop(from=0, until=line.length) { i =>
+          if (line(i) == delimiter) count += 1
+        }
+        count
+      }
+      if (numExpected > numFields) {
+        assert(fields.length < numExpected + fields.map(_.length).sum)
+        fields = Array.fill(numExpected)("")
+        assert(StringUtil.split(line, delimiter, fields) == numExpected)
+      }
```

Comment:
This is a lot of work to re-use the same array and keep expanding it.  Is it really worth it?  Especially since it requires you to iterate over the input line character-by-character twice for each line.
If you really want to do something like this I would suggest replacing with something like:

```scala
var fields = Array[String](100)
in.foreach { line => 
  var numFields   = StringUtil.split(line, delimiter, fields)
  while (numFields == fields.length) {
    fields = Array[String](fields.length * 2)
    numFields = StringUtil.split(line, delimiter, fields)
  }
```

Better though would be to extend that `split` method to have an option to place _all remaining text_ into the last value in the array instead of discarding extra values.  Then you could just size your array to `max(columns) + 2` and call it a day.

Reply:
https://github.com/fulcrumgenomics/commons/pull/64

Reply:
I suppose I'm over-optimizing here...